browser/src/Services/Sessions/SessionManager.ts

```diff
@@ -0,0 +1,102 @@
+import * as fs from "fs-extra"
+import { Commands, EditorManager } from "oni-api"
+import * as path from "path"
+
+import { SidebarManager } from "../Sidebar"
+import { SessionActions, SessionsPane, store } from "./"
+import { getUserHome } from "./../../Platform"
+
+export interface ISession {
+    name: string
+    id: string
+    file: string
+    directory: string
+    updatedAt: number
+}
+
+export interface ISessionService {
+    sessionsDir: string
+    sessions: ISession[]
+    persistSession(sessionName: string): Promise<ISession>
+    restoreSession(sessionName: string): Promise<ISession>
+}
+
+/**
+ * Class SessionManager
+ *
+ * Provides a service to manage oni session i.e. buffers, screen layout etc.
+ *
+ */
+export class SessionManager implements ISessionService {
+    private _store = store({ sessionManager: this, fs })
+
+    constructor(
+        private _editorManager: EditorManager,
+        private _sidebarManager: SidebarManager,
+        private _commands: Commands.Api,
+    ) {
+        fs.ensureDirSync(this.sessionsDir)
+        this._sidebarManager.add(
+            "save",
+            new SessionsPane({ store: this._store, commands: this._commands }),
+        )
+    }
+
+    public get sessions() {
+        return this._store.getState().sessions
+    }
+
+    public get sessionsDir() {
+        return path.join(getUserHome(), ".config", "oni", "sessions")
```

Comment:
I think you'll want to use `getUserConfigFolderPath` here, which gets the `oni` folder directly.

On Windows the `oni` folder is in `%APPDATA%\oni` versus `~/.config/oni`.
May also help testing.

https://github.com/onivim/oni/blob/e46c720fc82f1466347016bc0916a8d744c94793/browser/src/Services/Configuration/UserConfiguration.ts#L27-L40